block_model.rb

```diff
@@ -0,0 +1,73 @@
+class BlockModel
+  attr_reader :blocks
+
+  def initialize(blocks)
+    @blocks = blocks
+  end
+
+  def reblock(rx, ry, rz)
+    new_blocks = []
+    nx = @blocks.length / rx
+    ny = @blocks.length[0] / ry
+    nz = @blocks.length[0][0] / rz
+    for i in (0..nx)
+      new_blocks[i] = []
+      for j in (0..ny)
+        new_blocks[i][j] = []
+        for k in (0..nz)
+          new_blocks[i][j][k] = {}
+        end
+      end
+    end
+
```

Comment:
In this empty space of code, there could be a comment or region describing what was accomplished up to this point in the algorithm, or will be attempted in the next seccion.